src/request_collector/store_monitor_request.py

```diff
@@ -40,27 +42,26 @@ def _run(self):
                 ))
                 return False
 
-        self.state_db.store_monitor_request(self.msg)
+        self.state_db.upsert_monitor_request(self.msg)
         return True
 
-    def check_channel(self, monitor_request: MonitorRequest, channel: sqlite3.Row):
+    def check_channel(self, monitor_request: MonitorRequest, channel: Channel):
         """We must know about the channel and it must be open"""
-        return channel is not None and channel['state'] == ChannelState.OPENED
+        return channel is not None and channel.state == ChannelState.OPENED
```

Comment:
I think we no longer have this constraint, as we said that we allow to provide MRs before the channel is opened (maybe due to confirmations)  and after it was closed

Reply:
Right. In the long run we need tests that exercise these cases.

Reply:
Yes, we need to state these assumptions somewhere, test them (in the MS part this is done already) and also integrate the request collector in the end to end test.